shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -1511,14 +1514,33 @@ void updateSemantics(
       // of the screen in landscape mode. We must handle the translation ourselves for the
       // a11y nodes.
       if (Build.VERSION.SDK_INT >= 23) {
-        WindowInsets insets = rootAccessibilityView.getRootWindowInsets();
-        if (insets != null) {
-          if (!lastLeftFrameInset.equals(insets.getSystemWindowInsetLeft())) {
-            rootObject.globalGeometryDirty = true;
-            rootObject.inverseTransformDirty = true;
+        boolean needsToApplyCutoutInsect = true;
+        if (Build.VERSION.SDK_INT >= 28) {
+          Context context = rootAccessibilityView.getContext();
+          if (context instanceof Activity) {
```

Comment:
I am not sure what is the best way to get the window attributes, if there is a better way, please let me know

Reply:
this works, but what about also handling `ContextWrapper`?  Maybe, it could use https://github.com/flutter/engine/blob/40edba381e758e4abaaa4e59192748f12eb97895/shell/platform/android/io/flutter/view/FlutterView.java#L260